lib/util.js

```diff
@@ -403,6 +403,92 @@ module.exports = {
     };
   },
 
+  /**
+   * Generates object containing objects of tags and their requests
+   * @param {object} spec - openAPI spec object
+   * @returns {object} returns an object containing objects of tags and their requests
+   */
+  generateFolderObjectFromTags: function(spec) {
+    var globalTags = spec.tags,
+      localTags = [],
+      globalTag = '',
+      paths = spec.paths,
+      currentPathObject = {},
+      operationObject = {},
+      folderObj = {},
+      collectionVariables = {},
+      commonParams = '',
+      summary = '',
+      pathLevelServers = '',
+      name = '',
+      description = '';
+
+    // checking if tags are defined at the root level
+    if (globalTags.length > 0) {
+      for (let tag in globalTags) {
+        if (globalTags.hasOwnProperty(tag)) {
+          globalTag = globalTags[tag];
+          name = globalTag.name;
+          if (globalTag.hasOwnProperty('description')) {
+            description = globalTag.description;
+          }
+          // adding an object in folder object with tag name as key
+          folderObj[name] = {
+            description: description,
+            requests: []
+          };
+        }
+      }
+    }
+
+    for (let path in paths) {
+      if (paths.hasOwnProperty(path)) {
+        currentPathObject = paths[path];
+        for (let operation in currentPathObject) {
+          if (currentPathObject.hasOwnProperty(operation)) {
+            operationObject = currentPathObject[operation];
+            localTags = operationObject.tags;
+            // eslint-disable-next-line max-depth
```

Comment:
we're using the eslint-disable thing too much
We should default to no-tag while creating the value in the first place